test/js-native-api/test_object/test_object.c

```diff
@@ -106,6 +106,62 @@ static napi_value GetSymbolNames(napi_env env, napi_callback_info info) {
   return output;
 }
 
+static napi_value GetWritableNames(napi_env env, napi_callback_info info) {
+  size_t argc = 1;
+  napi_value args[1];
+  NODE_API_CALL(env, napi_get_cb_info(env, info, &argc, args, NULL, NULL));
+
+  NODE_API_ASSERT(env, argc >= 1, "Wrong number of arguments");
+
+  napi_valuetype value_type0;
+  NODE_API_CALL(env, napi_typeof(env, args[0], &value_type0));
+
+  NODE_API_ASSERT(
+      env,
+      value_type0 == napi_object,
+      "Wrong type of arguments. Expects an object as first argument.");
+
+  napi_value output;
+  NODE_API_CALL(
+      env,
+      napi_get_all_property_names(env,
+                                  args[0],
+                                  napi_key_include_prototypes,
+                                  napi_key_enumerable | napi_key_writable,
```

Comment:
Nonblocking: I think we can split out the `napi_key_enumerable` to a new method to test we can actually get non-enumerable keys.

Reply:
@legendecas, it is a good point! I did it originally that way, but the issue was that it returns too many properties such as all method names from `Object`. So, adding the `napi_key_enumerable` was the simplest way to work around it. Though, I can probably do it for the current object only without including prototypes.